yarn-project/acir-simulator/src/avm/avm_machine_state.ts

```diff
@@ -1,9 +1,17 @@
 import { Fr } from '@aztec/foundation/fields';
 
+import { AvmExecutionEnvironment } from './avm_execution_environment.js';
+
 /**
  * Store's data for an Avm execution frame
  */
 export class AvmMachineState {
+  /**
+   * Execution environment contains hard coded information that is received from the kernel
+   * Items like, the block header and global variables fall within this category
+   */
+  public readonly executionEnvironment: AvmExecutionEnvironment;
+
   /** - */
   public readonly calldata: Fr[];
```

Comment:
Calldata seems to be both here and in the environment, is that expected?

Reply:
In conversations with David I originally deemed that it should be separate from the execution environment, but the more I think about it, it fits the readonly nature of the rest of executionEnvironment, so i will move it in and push